packages/arb-token-bridge-ui/src/util/deposits/helpers.ts

```diff
@@ -110,14 +111,66 @@ export const updateAdditionalDepositData = async ({
     })
   }
 
-  // finally, else if the transaction is not ETH ie. it's a ERC20 token deposit
-  return updateTokenDepositStatusData({
+  let { value2 } = depositTx
+
+  // value2 not found in local storage, we fetch it instead
+  if (!value2) {
+    try {
+      // Get maxSubmissionCost, which is the amount of ETH sent in batched ERC-20 deposit + max gas cost
+      const maxSubmissionCost = utils.formatEther(
+        (
+          l1ToL2Msg as ParentToChildMessageReader
+        ).messageData.maxSubmissionFee.toString()
+      )
+
+      const gasCost = await getGasCostOnChildChain({
+        l1ToL2Msg: l1ToL2Msg as ParentToChildMessageReader
+      })
+
+      if (!gasCost) {
+        throw 'failed to calculate gas cost, value2 will be undefined'
+      }
+
+      value2 = String(Number(maxSubmissionCost) - Number(gasCost))
+
+      if (Number(value2) < 0.0001) {
```

Comment:
Nice. Maybe we definte this as a config variable? `MAX_SUBMISSION_FEE_THRESHOLD` or something like that?